leakpro/attacks/utils/shadow_model_handler.py

```diff
@@ -304,3 +305,51 @@ def identify_models_trained_on_samples(self:Self, shadow_model_indices: list[int
                     shadow_model_trained_on_data_index[i, j] = sample_indices[j] in train_indices
 
         return shadow_model_trained_on_data_index
+
+    def _load_metadata(self:Self, index:int) -> Module:
+        """Load a shadow model from a saved state.
+
+        Args:
+        ----
+            index (int): The index of the shadow model to load.
+
+        Returns:
+        -------
+            Module: The loaded shadow model.
+
+        """
+        if index < 0:
+            raise ValueError("Index cannot be negative")
+        if index >= len(os.listdir(self.storage_path)):
+            raise ValueError("Index out of range")
+        meta_data = {}
+        with open(f"{self.storage_path}/{self.metadata_storage_name}_{index}.pkl", "rb") as f:
+            meta_data = pickle.load(f)
+        return meta_data
+        
+    def get_shadow_model_metadata(self:Self, num_models:int) -> list:
+        """Load the the shadow model metadata."""
+        metadata = []
+        for i in range(num_models):
```

Comment:
Yes, but it will load the metadata after the all shadowmodels have been loaded/trained. Maybe I am missing something here but this process will be the same whether or not all shadow models or some is trained or loaded from disc

Reply:
yea, let's keep this for now.